crowbar_framework/app/models/service_object.rb

```diff
@@ -1084,6 +1084,17 @@ def apply_role(role, inst, in_queue)
     element_order = new_deployment["element_order"]
 
     #
+    # stop automatic chef-client on all nodes
+    #
+    nodes_to_freeze_chef_service = new_elements.values.flatten
+    nodes_to_freeze_chef_service.each do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      wait_for_chef_clients(node)
```

Comment:
I just saw the `wait_for_chef_clients` method has an `exit(1)` after a timeout, would this be acceptable in this case as well?